### ISSUE TITLE
Can't import link-summoner for a NextJS instance

### ISSUE THREAD

**Issue by akrolsmir** [Stvad/link-summoner#3]
Hey Vlad! Trying out your instructions to add link-summoner as an NPM package, but I'm seeing this error:
![image](https://user-images.githubusercontent.com/856034/168424612-3eddd866-24ff-4891-a512-2cfccdc906d4.png)

Even though `initPreviews()` is wrapped to be [only called client side](https://github.com/manifoldmarkets/manifold/pull/150/commits/15f076a112872fca044d38f4eef422ae9c99aad9):
```
  useEffect(() => {
    if (typeof document !== 'undefined') {
      // This seems to work...
      console.log(document)
      // But not this...
      initPreviews()
    }
  }, [])
```

Pretty sure it's because initShadowRoot is called [here](https://github.com/Stvad/link-summoner/blob/4f73e8215fac7e79a293fd1b7bcd3ddc958dbda1/src/tippy.ts#L5) on initial import, so we try to run with document while NextJS is running server-side logic. 

Could we refactor this so initShadowRoot only gets called after calling initPreview()?

**Comment by Stvad** [comment]
ah, that would do it. I wonder if there is a good way to test this on my side 🤔. will push an update to lazy init shadowroot later!

**Comment by Stvad** [comment]
(you don't need to wrap this into `useEffect` btw, though I imagine you've tried when initial document check didn't work..)

**Comment by Stvad** [comment]
fixed in 1.3. also tested importing the result in nodejs runtime, which seems to work, which should presumably translate to it working in nextjs ssr env
